source/extensions/filters/listener/proxy_protocol/proxy_protocol.h

```diff
@@ -43,6 +43,17 @@ using ConfigSharedPtr = std::shared_ptr<Config>;
 
 enum ProxyProtocolVersion { Unknown = -1, InProgress = -2, V1 = 1, V2 = 2 };
 
+enum class ProxyState{
+    //failed to read proxy protocol
+    PrtclFail,
+
+    //Unsupported command or address family or transport
```

Comment:
Can you document what the usage of this is? it looks like it is only used in one method, and the try/catch that's replaced only tests for `PrtclFail`.

Reply:
I believe ProtocolFail (changed it from PrtclFail following your next comment) enum value is used in multiple methods like onRead(), onReadWorker(), parseV1Header, and so on. OR, are you referring to something else? 

The try/catch that's replaced only tests for ProtocolFail as it was supposed to throw an exception if there was this error only. Is there any change that you would like to recommend?